src/Agent.Worker/Variables.cs

```diff
@@ -312,6 +339,26 @@ public string Get(string name)
             return null;
         }
 
+        public VariableScope CreateScope()
+        {
+            return new VariableScope(this);
```

Comment:
I think the crux of my question is are you expecting `this` to be passed by reference here? My understanding was that it would be passed by value so any changes won't be seen in this object itself. I think I might just not know enough about how C# handles variables though.

Reply:
you are correct. I am assuming it to be passed by reference here. I have a unit test that I thought proved this worked, but now I am doubting my little understanding of C#.

Reply:
I'm not sure that it's **not** how it works to be clear. Normally variables wouldn't be passed by reference here, I think `this` might be a reference type itself though? I just don't know enough here to be sure, and am struggling to find the right info

Reply:
i think pointers that point to cs reference types would be passed by value here. 
In a real language the constructor would be: 
```c++
VariableScope( Variables* data ) {};
```
in member functions of the `Variables` class, `this` would be of type `Variables*` which can be pass-by-value into the ctor.